posts/2024-11-19-24.0.0.12-beta.adoc

```diff
@@ -0,0 +1,234 @@
+4.0] (`xmlWS-3.0`)
+* xref:{url-prefix}/docs/latest/reference/feature/jaxws-2.2.html[Java Web Services] (`jaxws-2.2`)
+
+You can apply the configuration to web services (`webService`) or web service clients (`webServiceClient`), either globally, or to an individual client or web service implementation.
+
+The examples in the following sections show how to enable each option. To disable any of these options, set the corresponding attributes to `false`.
+
+=== XML schema validation
+
+You can set the `enableSchemaValidation=true` attribute to provide more insight into JAXB unmarshalling exceptions and make painful message debugging easier. This option is the highest level of XML validation, which provides faster debugging and the most thorough checks on inbound message contents. But it comes with a tradeoff: higher performance cost.
+
+==== Global XML schema validation
+
+The following example shows how to enable XML schema validation for web services globally for your Open Liberty runtime:
+
+[source,xml]
+----
+<webService enableSchemaValidation="true" />
+----
+
+To enable XML schema validation globally for web service clients, set the same attribute on the `webServiceClient` element:
+
+[source,xml]
+----
+<webServiceClient enableSchemaValidation="true" />
+----
+
+==== Targeted XML schema validation
+
+The following example shows how to enable XML schema validation for a particular web service by using the web service port:
+
+[source,xml]
+----
+<webService portName="<web service port name>"  enableSchemaValidation="true" />
+----
+
+The value of `portName` is the port name of the Web Service implementation you're configuring.  This name comes from your `@WebService(portName=<web service port name>` annotated class.
+Alternatively, you can check the `<wsdl:port ... name="Web Service Port Name">` line in your WSDL file for the port name.
+
+
+The following example shows how to enable XML schema validation for a specific client service:
+
+[source,xml]
+----
+<webServiceClient serviceName="<client service name>"  enableSchemaValidation="true" />
+----
+
+The value of `serviceName` is the name of the Web Service Client you're configuring. This name comes from your `@WebServiceClient(serviceName=<client service name>` annotated stub class for managed clients.
+For unmanaged clients, you can check the `<wsdl:service name="<client service name>"` line in your WSDL file.
+
+
+
+=== Default validation
+
+You can configure the default level of JAXB validation of inbound SOAP Messages with the `enableDefaultValidation=true` attribute. Depending on whether your application uses an XML Web Services client or web service implementation, Open Liberty either enables or disables default validation. Default validation is much more efficient than XML schema validation, so enabling it provides basic message validation with lower overhead. Disabling it lets you ignore various unmarshalling errors for problematic messages.
```

Comment:
> Depending on whether your application uses an XML Web Services client or web service implementation, Open Liberty either enables or disables default validation.

does this mean we enable default validation for clients and disable it for services?